samples/subsys/usb/hid/src/main.c

```diff
@@ -81,36 +85,57 @@ int get_report_cb(struct usb_setup_packet *setup, s32_t *len,
 	return 0;
 }
 
+static void send_report(struct k_work *work)
+{
+	static u8_t report_1[2] = { REPORT_ID_1, 0x00 };
+	int ret, wrote;
+
+	ret = hid_int_ep_write(report_1, sizeof(report_1), &wrote);
+
+	SYS_LOG_DBG("Wrote %d bytes with ret %d", wrote, ret);
+
+	/* Increment reported data */
+	report_1[1]++;
+}
+
+static void in_ready_cb(void)
+{
+	k_delayed_work_submit(&delayed_report_send, REPORT_TIMEOUT);
```

Comment:
Can it happen that this function will be called again before REPORT_TIMEOUT time passes?